File: BackEnd/atwom_proj/src/main/java/com/a2m/library/dto/CheckoutDTO.java
package com.a2m.library.dto;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

import com.a2m.library.constant.CheckoutStatus;

@Data
public class CheckoutDTO{
    private Integer id;
    private Long userUid;
    private UserDTO user;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private CheckoutStatus status;
    private List<CheckoutDetailDTO> checkoutDetails;
}

File: FrontEnd/admin/src/app/model/CheckoutDTO.ts
import { UserDTO } from './UserDTO';
import { CheckoutStatus } from './CheckoutStatus';
import { CheckoutDetailDTO } from './CheckoutDetailDTO';

export interface CheckoutDTO {
  id: number;
  user: UserDTO;
  userUid: number;
  userFullName: string;
  startTime: string;
  endTime: string;
  expiredTime: string;
  fine: number;
  status: CheckoutStatus;
  checkoutDetails: CheckoutDetailDTO[];
}

export { CheckoutStatus };
